src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMassesGenerator.cpp

```diff
@@ -224,86 +249,112 @@ namespace OpenMS
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet3));
-
-        std::vector<double> doublet4(1, 0);
-        doublet4.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(1));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet4));
-
-        std::vector<double> doublet5(1, 0);
-        doublet5.push_back(mass_pattern_list_[i].getMassShiftAt(3) - mass_pattern_list_[i].getMassShiftAt(1));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet5));
-
-        std::vector<double> doublet6(1, 0);
-        doublet6.push_back(mass_pattern_list_[i].getMassShiftAt(3) - mass_pattern_list_[i].getMassShiftAt(2));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet6));
+        MultiplexDeltaMasses doublet1;
+        doublet1.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(0));
+        doublet1.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(1));
+        mass_pattern_list_.push_back(doublet1);
+
+        MultiplexDeltaMasses doublet2;
+        doublet2.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(0));
+        doublet2.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(2));
+        mass_pattern_list_.push_back(doublet2);
+
+        MultiplexDeltaMasses doublet3;
+        doublet3.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(0));
+        doublet3.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(3));
+        mass_pattern_list_.push_back(doublet3);
+
+        MultiplexDeltaMasses doublet4;
+        doublet4.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(1));
+        doublet4.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(2));
+        mass_pattern_list_.push_back(doublet4);
+
+        MultiplexDeltaMasses doublet5;
+        doublet5.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(1));
+        doublet5.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(3));
+        mass_pattern_list_.push_back(doublet5);
+
+        MultiplexDeltaMasses doublet6;
+        doublet6.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(2));
+        doublet6.addDeltaMass(mass_pattern_list_[i].getDeltaMassAt(3));
+        mass_pattern_list_.push_back(doublet6);
       }
 
       // add singlets
-      std::vector<double> singlet(1, 0);
-      mass_pattern_list_.push_back(MultiplexDeltaMasses(singlet));
+      MultiplexDeltaMasses dm;
+      dm.addDeltaMass(0,"any_label");
+      mass_pattern_list_.push_back(dm);
     }
     else if (n > 4)
     {
       throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, __PRETTY_FUNCTION__, "Knock-outs for multiplex experiments with more than 4 samples not supported.");
     }
+    
+    // sort mass patterns
+    // (from small mass shifts to larger ones, i.e. few miscleavages = simple explanation first)
+    std::sort(mass_pattern_list_.begin(), mass_pattern_list_.end());
+
   }
   
   void MultiplexDeltaMassesGenerator::printLabelsList() const
```

Comment:
usually its more flexible to use an ofstream as input argument and write into it.
This way, the user can decide to write to any stream, not just cout.